tests/test_value_parser.py

```diff
@@ -80,8 +80,9 @@ def test_ua_graph_xml_encode_for_eurange(ua_graph):
     )
 
     path_to_tests = get_project_root() / "tests"
+    # Technically, the eurange_lines is not a valid XML file, so it is saved as a TXT file.
```

Comment:
A commit hook for checking XML files spotted this invalid XML file (missing root tag)! It means that the `write_nodeset()` method is able to generate XMLs with invalid elements. @olavpe Maybe we should investigate it at some point in the future?

Reply:
Oh wow! That is not good. I am very glad you were able to catch this. Luckily, I have not heard that it has stopped US in our work, but regardless this should be fixed. I can set up a task for it.